src/network/networkd-sysctl.c

```diff
@@ -40,10 +42,82 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static void sysctl_overwrite_check(sd_event_source *source, const struct inotify_event *event, Hashmap *sysctls) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        const char *dir_path, *value;
+        size_t len;
+        int r;
+
+        assert(event);
+
+        r = sd_event_source_get_inotify_path(source, &dir_path);
+        if (r < 0)
+                return;
+
+        path = path_join(dir_path, event->name);
+        if (!path)
+                return;
+
+        value = hashmap_get(sysctls, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return;
+
+        if (read_full_file(path, &buf, &len) < 0)
+                return;
+
+        delete_trailing_chars(buf, "\n");
+
+        if (!streq(value, buf))
+                log_warning("sysctl %s was changed from '%s' to '%s'", path, value, buf);
+}
+
+static int sysctl_inotify_manager(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Manager *manager = ASSERT_PTR(userdata);
+
+        sysctl_overwrite_check(source, event, manager->sysctls);
+
+        return 0;
+}
+
+static int sysctl_inotify_link(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        Link *link = ASSERT_PTR(userdata);
+
+        sysctl_overwrite_check(source, event, link->sysctls);
+
+        return 0;
+}
+
+static int manager_inotify_add(Manager *manager) {
+        sd_event_source **source;
+
+        assert(manager);
+
+        source = manager->sysctl_event_sources;
+
+        FOREACH_STRING(p,
+                "/proc/sys/net/ipv4/conf/all",
+                "/proc/sys/net/ipv4/conf/default",
+                "/proc/sys/net/ipv6/conf/all",
+                "/proc/sys/net/ipv6/conf/default") {
+                int r;
```

Comment:
`int r;` is always special to us. It's the variable that we always write error codes to. Hence it gets this special privilege in our codebase that it is always defined in outermost function scope. i.e. please move this one block up